core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -103,6 +104,9 @@ object SparkSubmit {
    /___/ .__/\_,_/_/ /_/\_\   version %s
       /_/
                         """.format(SPARK_VERSION))
+    printStream.println("Branch %s".format(SPARK_BRANCH))
+    printStream.println("Compiled by user %s on %s".format(SPARK_BUILD_USER, SPARK_BUILD_DATE))
+    printStream.println("Url %s".format(SPARK_REPO_URL))
```

Comment:
It would be great to also add the build info to org.apache.spark.repl.SparkILoop